packages/eslint-plugin/lib/config/rules/typescript.js

```diff
@@ -131,6 +131,19 @@ module.exports = {
       selector: 'typeLike',
```

Comment:
I used the `--print-config` CLI arg to log out the eslint rules our repo used before the recent changes

I confirmed that the rules for `typescript-eslint/naming-convention` matched those in this file, so merging the two rule sets together seems like our best bet
```
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "default",
        "format": [
          "camelCase",
          "PascalCase",
          "UPPER_CASE"
        ],
        "leadingUnderscore": "allow",
        "trailingUnderscore": "allow"
      },
      {
        "selector": "default",
        "filter": {
          "match": true,
          "regex": "^(__|UNSAFE_).+$"
        },
        "format": null
      },
      {
        "selector": "typeLike",
        "format": [
          "PascalCase"
        ]
      }
    ],
```